mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1506,6 +1506,109 @@ impl TakerSwap {
         }
     }
 
+    /// Sets up the watcher reward for the taker's payment in the swap.
+    ///
+    /// The reward mainly serves as compensation to watchers for the mining fees
+    /// paid to execute the transactions.
+    ///
+    /// The reward configuration depends on the specific requirements of the coins
+    /// involved in the swap.
+    /// Some coins may not support watcher rewards at all.
+    async fn setup_watcher_reward(&self, taker_payment_lock: u64) -> Result<Option<WatcherReward>, String> {
+        if !self.r().watcher_reward {
+            return Ok(None);
+        }
+
+        let reward_amount = self.r().reward_amount.clone();
+        self.taker_coin
+            .get_taker_watcher_reward(
+                &self.maker_coin,
+                Some(self.taker_amount.clone().into()),
+                Some(self.maker_amount.clone().into()),
+                reward_amount,
+                taker_payment_lock,
+            )
+            .await
+            .map(Some)
+            .map_err(|err| ERRL!("Watcher reward error: {}", err.to_string()))
+    }
+
+    /// Processes watcher-related logic for the swap by preparing and broadcasting necessary data.
+    ///
+    /// This function creates spend/refund preimages and broadcasts them to watchers if both coins
+    /// support watcher functionality and watchers are enabled.
+    ///
+    /// The preimages allow watchers to either complete the swap by spending the maker payment
+    /// or refund the taker payment if needed.
+    async fn process_watcher_logic(&self, transaction: &TransactionEnum) -> Option<TakerSwapEvent> {
+        let watchers_enabled_and_supported = self.ctx.use_watchers()
+            && self.taker_coin.is_supported_by_watchers()
+            && self.maker_coin.is_supported_by_watchers();
+
+        if !watchers_enabled_and_supported {
+            return None;
+        }
+
+        let maker_payment_spend_preimage_fut = self.maker_coin.create_maker_payment_spend_preimage(
+            &self.r().maker_payment.as_ref().unwrap().tx_hex,
+            self.maker_payment_lock.load(Ordering::Relaxed),
+            self.r().other_maker_coin_htlc_pub.as_slice(),
+            &self.r().secret_hash.0,
+            &self.unique_swap_data()[..],
+        );
+
+        let time_lock = match std::env::var("USE_TEST_LOCKTIME") {
```

Comment:
Should we not allow this test locktime only if enabled by a cfg feature?
(I guess we tend to make all test code cfg enabled)